core/aws_ddk_core/stages/firehose_s3.py

```diff
@@ -0,0 +1,179 @@
+        error_output_prefix: Optional[str] = None,
+        logging: Optional[bool] = True,
+        log_group: Optional[ILogGroup] = None,
+    ) -> None:
+        """
+        DDK Firehose to S3 stage.
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the stage
+        environment_id : str
+            Identifier of the environment
+        bucket_name: Optional[str] = None
+            Name of S3 Bucket to be created as a delivery destination
+        delivery_stream_name: Optional[str] = None
+            Name of the Firehose Delivery Stream
+        delivery_stream: Optional[firehose.IDeliveryStream] = None
+            Existing Delivery Stream to use in this stage
+        bucket: Optional[IBucket] = None
+            Existing S3 Bucket to use as a destination for the Firehose Stream
+        buffering_interval: Optional[Duration] = None
+            The length of time that Firehose buffers incoming data before delivering it to the S3 bucket.
+            Minimum: Duration.seconds(60)
+            Maximum: Duration.seconds(900)
+            Default: Duration.seconds(300)
+        buffering_size: Optional[Size] = None
+            The size of the buffer that Kinesis Data Firehose uses for incoming
+            data before delivering it to the S3 bucket.
+            Minimum: Size.mebibytes(1)
+            Maximum: Size.mebibytes(128)
+            Default: Size.mebibytes(5)
+        compression: Optional[Compression] = None
+            The type of compression that Kinesis Data Firehose uses to compress
+            the data that it delivers to the Amazon S3 bucket.
+            Default: - UNCOMPRESSED
+        data_output_prefix: Optional[str] = None
+            A prefix that Kinesis Data Firehose evaluates and adds to records before writing them to S3.
+            This prefix appears immediately following the bucket name.
+            Default: “YYYY/MM/DD/HH”
+        encryption_key: Optional[IKey] = None
+            The AWS KMS key used to encrypt the data delivered to your Amazon S3 bucket
+        error_output_prefix: Optional[str] = None
+            prefix that Kinesis Data Firehose evaluates and adds to failed records before writing them to S3.
+            This prefix appears immediately following the bucket name.
+            Default: “YYYY/MM/DD/HH”
+        logging: Optional[bool] = True
+            If true, log errors when data transformation or data delivery fails.
+            If `log_group` is provided, this will be implicitly set to true.
+            Default: true - errors are logged.
+        log_group: Optional[ILogGroup] = None
+            The CloudWatch log group where log streams will be created to hold error logs.
+            Default: - if logging is set to true, a log group will be created for you.
+
+        """
+        super().__init__(scope, id)
```

Comment:
@kukushking I suppose we would want to consider whether we wish to also front this stage with an `optional` kinesis data stream too like in the original architecture diagram or not...

Reply:
Lost the link - can you share, please? If that's a popular use case I don't see why not

Reply:
https://aws.amazon.com/solutions/implementations/aws-streaming-data-solution-for-amazon-kinesis/
Option #2 where we can add an optional data stream fronting the Firehose delivery stream
A bit like you can create an option DLQ in our SQSToLambda stage